lib/sycamore/sycamore/tests/integration/connectors/opensearch/test_opensearch_read.py

```diff
@@ -110,3 +115,89 @@ def test_ingest_and_read(self, setup_index, exec_mode):
 
         for i in range(len(doc.elements) - 1):
             assert doc.elements[i].element_index < doc.elements[i + 1].element_index
+
+    def _test_ingest_and_read_via_docid_reconstructor(self, setup_index, exec_mode, cache_dir):
+        """
+        Validates data is readable from OpenSearch, and that we can rebuild processed Sycamore documents.
+        """
+
+        print(f"Using cache dir: {cache_dir}")
+
+        def doc_reconstructor(index_name: str, doc_id: str) -> Document:
+            prefix = f"doc-{doc_id}"
+            found = None
+            files = os.listdir(cache_dir)
+            for f in files:
+                if f.startswith(prefix):
+                    found = f
+                    break
+            assert found, "Doc not found in cache"
+            import pickle
+
+            data = pickle.load(open(f"{cache_dir}/{found}", "rb"))
+            return Document(**data)
+
+        path = str(TEST_DIR / "resources/data/pdfs/Ray.pdf")
+        context = sycamore.init(exec_mode=exec_mode)
+        original_docs = (
+            context.read.binary(path, binary_format="pdf")
+            .partition(partitioner=UnstructuredPdfPartitioner())
+            .materialize(cache_dir)
```

Comment:
This should not work; default materialize naming is content-hash+doc_id. 
https://github.com/aryn-ai/sycamore/blob/f80ff0e9480cd3ed48d98d74a235621f8fccd01c/lib/sycamore/sycamore/materialize.py#L313
You need to override the naming function to match with line 137.

Reply:
Ok